src/with-component.js

```diff
@@ -1,13 +1,18 @@
 import { h, render } from 'preact';
 
-import { HTMLElement } from './util';
+import { HTMLElement, sym } from './util';
 import { withProps } from './with-props';
 import { withRender } from './with-render';
 import { withUnique } from './with-unique';
 
+const _preactDom = sym('_preactDom');
+
 export const withComponent = (Base = HTMLElement) => class extends withUnique(withRender(withProps(Base))) {
   rendererCallback (shadowRoot, renderCallback) {
-    render(renderCallback(), shadowRoot);
+    const dom = render(renderCallback(), shadowRoot, this[_preactDom]);
+    if (!this[_preactDom]) {
```

Comment:
That's fair too, I hadn't thought about size. I have no problem changing it. I am curious which is better thought, probably you're right that the branching is worse. I don't know why my instinct is to add the diff.